InnerEye/ML/SSL/lightning_containers/ssl_container.py

```diff
@@ -0,0 +1,243 @@
+#  ------------------------------------------------------------------------------------------
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+#  Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+#  ------------------------------------------------------------------------------------------
+import logging
+from dataclasses import dataclass
+from enum import Enum
+from pathlib import Path
+from typing import Any, Dict, Optional, Tuple, Union
+
+import param
+from pytorch_lightning import LightningModule
+from yacs.config import CfgNode
+
+from InnerEye.ML.SSL.datamodules_and_datasets.cifar_datasets import InnerEyeCIFAR10, InnerEyeCIFAR100
+from InnerEye.ML.SSL.datamodules_and_datasets.cxr_datasets import CheXpert, NIH, RSNAKaggleCXR
+from InnerEye.ML.SSL.datamodules_and_datasets.datamodules import CombinedDataModule, InnerEyeVisionDataModule
+from InnerEye.ML.SSL.datamodules_and_datasets.transforms_utils import InnerEyeCIFARLinearHeadTransform, \
+    InnerEyeCIFARTrainTransform, \
+    get_cxr_ssl_transforms
+from InnerEye.ML.SSL.encoders import get_encoder_output_dim
+from InnerEye.ML.SSL.lightning_modules.byol.byol_module import BYOLInnerEye
+from InnerEye.ML.SSL.lightning_modules.simclr_module import SimCLRInnerEye
+from InnerEye.ML.SSL.lightning_modules.ssl_online_evaluator import SSLOnlineEvaluatorInnerEye
+from InnerEye.ML.SSL.utils import SSLDataModuleType, SSLTrainingType, load_ssl_augmentation_config
+from InnerEye.ML.lightning_container import LightningContainer
+
+
+@dataclass
+class DataModuleArgs:
+    augmentation_params: Optional[CfgNode]
+    dataset_name: str
+    dataset_path: Optional[Path]
+    batch_size: int
+
+
+class EncoderName(Enum):
+    resnet18 = "resnet18"
+    resnet50 = "resnet50"
+    resnet101 = "resnet101"
+    densenet121 = "densenet121"
+
+
+class SSLDatasetName(Enum):
+    RSNAKaggle = "RSNAKaggle"
+    NIH = "NIH"
+    CIFAR10 = "CIFAR10"
+    CIFAR100 = "CIFAR100"
+    CheXpert = "CheXpert"
+
+
+InnerEyeDataModuleTypes = Union[InnerEyeVisionDataModule, CombinedDataModule]
+
+
+class SSLContainer(LightningContainer):
+    """
+    This container is the based module to train an SSL model (either using BYOL or SimCLR).
```

Comment:
What are the other hidden parameters that users need to specify in here?
e.g. `ssl_training_lr`, `num_epochs` ...

How is the user supposed to know these parameters?

Reply:
I summarized all the parameters in the docs.

Reply:
I can add a pointer to the docs in the docstring